src/panoramic/cli/transform/executor.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from dataclasses import dataclass
+
+from dbt.exceptions import FailedToConnectException
+from requests import RequestException
+
+from panoramic.cli.connections import (
+    CONNECTION_KEYS,
+    Connections,
+    get_dialect_credentials,
+)
+from panoramic.cli.errors import (
+    ConnectionNotFound,
+    TransformCompileException,
+    TransformExecutionFailed,
+)
+from panoramic.cli.print import echo_info
+from panoramic.cli.transform.client import TransformClient
+from panoramic.cli.transform.pano_transform import PanoTransform
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class TransformExecutor:
+
+    company_slug: str
+    transform: PanoTransform
+    compiled_query: str
+
+    @classmethod
+    def from_transform(cls, company_slug: str, transform: PanoTransform) -> 'TransformExecutor':
```

Comment:
Not sure about the naming here - `Class.from_*` usually implies something non-side-effecty.

One way to fix it: can we separate the act of compiling from the act of executing? Just an idea but something like:

```
# first we compile the transform
compiled_transforms = [TransformCompiler.compile(transform) for transform in transforms]
# user confirms
for ct in compiled_transforms:
    # allows re-use of executor/connection pooling etc.
    executor = get_executor(compiled_transform.connection)
    result = executor.execute(compiled_transform)
```

Reply:
Split in 57a5ddc